### PR TITLE
Domain management revamp: update domain list column styles in sidebar

### LINKED ISSUE #97390: Fix scrollbar being visible in domain overview tabs in some devices
In some devices, a scrollbar appears below the navigation tabs. From an initial investigation, it seems that it stems from some scroll-related settings in Mac.

We will fix it under this issue.

### PR DESCRIPTION
**Pull request by Imran92**
Related to https://github.com/Automattic/wp-calypso/issues/96554

## Proposed Changes
Updates the looks of the left side of the Domains preview

* Fix the font size, spacing and styling of the Tab title ("Domains")
* Fix the look and style of the button to add domains
* Fix search bar
* Fix spacings
* Remove column names when showing domains only in sidebar
* Fix hiding left column if size of tab view and smaller
* Fix paddings of preview pane in mobile view
* Fixes scrollbar showing up in tab

## Why are these changes being made?

* To revamp domain management

## Testing Instructions

- Go to wordpress.com/domains/manage
- Browse around, check inner functionalities, make sure you see no issues in how the table looks and works
- Open a site's wp-admin area, go to Upgrades -> Emails, browse around and make sure everything works as before
- Also click on a few domains to make sure everything is working as before.
- Now enable the feature flag by entering window.sessionStorage.setItem('flags', 'calypso/all-domain-management') in the - browser console. Reload the page
- Now click on different domains. They should open in the new layout now.
- Check that the left side of the new layout (the domains list) matches the design.
- When you open wordpress.com/sites and click a site, the left side panel opens. The domains side panel should also match the sites side panel

| Before | After |
|--------|--------|
| Screenshot 2024-12-07 at 1 50 39 AM | Screenshot 2024-12-07 at 3 06 44 AM |

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/domains/domain-management/domain-overview-pane/style.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+@media (max-width: 960px) {
```
Comment (m1r0) [inline client/my-sites/domains/domain-management/domain-overview-pane/style.scss]: What do you think of using the [media query mixins](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md#media-queries) here instead?

File: client/my-sites/domains/domain-management/list/style.scss [hunk 2]
```diff
@@ -327,4 +327,68 @@
...
+			top: 2px;
+			width: 18px;
+			height: 18px;
+			margin-top: -2px;
+		}
+	}
+}
+
+// Similar to the sites-overview, hide the domains-overview list on the left on smaller screens.
+@media (max-width: 1280px) {
```
Comment (m1r0) [inline client/my-sites/domains/domain-management/list/style.scss]: Same as the previous comment.

**Comment by m1r0** [comment]
Looks good overall! I have just a few small notes.

* There's some extra margin at the bottom of the list. It's visible when the list is scrollable.

image

* As a suggestion (could be done in another PR), WDYT of showing the meatballs menu (the 3 dots) on hover similar to `/sites`? To make it even more consistent, we could rotate the dots vertically.
* Another suggestion - it would be nice to have an active state for the selected domain in the sidebar (same as `/sites`).

File: client/my-sites/domains/domain-management/domain-overview-pane/style.scss (hunk 1 above)
Comment (Imran92) [reply client/my-sites/domains/domain-management/domain-overview-pane/style.scss]: Done it here https://github.com/Automattic/wp-calypso/pull/97193/commits/ddd028a0cb3a9c56036dda1335b93b61666b433b

File: client/my-sites/domains/domain-management/list/style.scss (hunk 2 above)
Comment (Imran92) [reply client/my-sites/domains/domain-management/list/style.scss]: Removing it in favor of this PR https://github.com/Automattic/wp-calypso/pull/97240

**Comment by Imran92** [comment]
> There's some extra margin at the bottom of the list. It's visible when the list is scrollable.

I've removed the space [here](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71). Have also changed the scroll behavior to scroll including the search bar, to match the behavior of /sites.

> As a suggestion (could be done in another PR), WDYT of showing the meatballs menu (the 3 dots) on hover similar to /sites? To make it even more consistent, we could rotate the dots vertically.

I've made the meatball menu vertical and show on hover as well.

> Another suggestion - it would be nice to have an active state for the selected domain in the sidebar (same as /sites).

Good suggestion, but I've intentionally not done that yet, because the selected site determination process will change significantly with DataViews. So to avoid re-works, I think it'll be better to do it once we have the DataViews

**Review by m1r0 — COMMENTED** [review]
>  I've made the meatball menu vertical and show on hover as well.

Looks good!

> I've removed the space [here](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71). Have also changed the scroll behavior to scroll including the search bar, to match the behavior of /sites.

This is fixed on big resolutions but I'm still seeing the white space below 1400px. 🤔 Any ideas why that might be?

https://github.com/user-attachments/assets/d6a700c4-3470-4434-9b84-381ede954711

On the same note, after the latest changes, I've noticed some white space around the sidebar items. It looks like the items are no longer full-width.

image

> Good suggestion, but I've intentionally not done that yet, because the selected site determination process will change significantly with DataViews. So to avoid re-works, I think it'll be better to do it once we have the DataViews

Good to know, thanks!

**Comment by Imran92** [comment]
> This is fixed on big resolutions but I'm still seeing the white space below 1400px. 🤔 Any ideas why that might be?

Thanks for finding it! I've fixed the issue here - https://github.com/Automattic/wp-calypso/pull/97193/commits/d5998ef3050396dbd84c7f70102d8e618e2ea9b8

> On the same note, after the latest changes, I've noticed some white space around the sidebar items. It looks like the items are no longer full-width.

Not sure how it's happening there, I could reproduce this spacing only below 960px. In that case, hopefully it won't be an issue because once the following PR gets merged, the domain list will be hidden below 1280px

https://github.com/Automattic/wp-calypso/blob/415cea642be3a6e565ff561f06bb7bb9dad6063c/client/my-sites/domains/domain-management/list/style.scss#L328-L335

**Comment by m1r0** [comment]
> Not sure how it's happening there, I could reproduce this spacing only below 960px.

Interesting. For me, it happens on all resolutions. I can also reproduce it on Firefox. I've tracked it down to [this change](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71#diff-6c5972cdc95feb1074312c8ec2c57e557993e7797e0244bc6e08a51c08ac5582R240-R241) (flex on `.main`). A fix that works for me is using `width: 100%` to `.domains-table`.

**Comment by Imran92** [comment]
> Interesting. For me, it happens on all resolutions (as seen in the video above). I can also reproduce it on Firefox. I've tracked it down to [this change](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71#diff-6c5972cdc95feb1074312c8ec2c57e557993e7797e0244bc6e08a51c08ac5582R240-R241) (flex on .main). A fix that works for me is using width: 100% to .domains-table.

I still can't reproduce it after many tries 🤔 Can you kindly also try the calypso.live link above? https://github.com/Automattic/wp-calypso/pull/97193#issuecomment-2524189772

**Comment by Imran92** [comment]
I've also fixed the issue https://github.com/Automattic/wp-calypso/issues/97390 here. So the following scrollbar scenario should not happen -

image

**Comment by m1r0** [comment]
> I still can't reproduce it after many tries 🤔 Can you kindly also try the calypso.live link above and see what happens?

It's even worse on calypso.live. The extra spacing is still there, but also the "Domains" header title is missing some spacing... 😞 I did a video in Firefox, but it happens in Chrome as well. Maybe ask someone else from the team to test it?

https://github.com/user-attachments/assets/70669eba-f322-46c5-b736-95434632d498

> I've also fixed the issue https://github.com/Automattic/wp-calypso/issues/97390 here. So the following scrollbar scenario should not happen -

I've [fixed](https://github.com/Automattic/wp-calypso/pull/97240/files#diff-8bb4f3f403bd6b260e3987bef17afc8f2318d3c0b702461dcb5c724e0b46adf9R40) the same thing. 😄 Let me know if I should remove it from my PR.

**Review by m1r0 — APPROVED** [review]
As we discussed, I fixed the issues that were popping up on my end and calypso.live.

@Imran92, could you give it a final look before merging?

**Comment by Imran92** [comment]
Thanks @m1r0 ! LGTM and merging

**Pull request merged**